File: app/src/test/java/org/jellyfin/androidtv/playback/overlay/CustomSeekProviderTest.java
package org.jellyfin.androidtv.playback.overlay;

import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;

public class CustomSeekProviderTest {

    @Test
    public void testGetSeekPositions_withSimpleDuration() {
        CustomSeekProvider customSeekProvider = new CustomSeekProvider(90_000);

        long[] expected = new long[]{0L, 30_000L, 60_000L, 90_000L};
        long[] actual = customSeekProvider.getSeekPositions();

        assertArrayEquals(expected, actual);
    }

    @Test
    public void testGetSeekPositions_withOddDuration() {
        CustomSeekProvider customSeekProvider = new CustomSeekProvider(130_000);

        long[] expected = new long[]{0L, 30_000, 60_000, 90_000, 120_000, 130_000};
        long[] actual = customSeekProvider.getSeekPositions();

        assertArrayEquals(expected, actual);
    }
}

File: app/src/test/kotlin/ui/playback/overlay/CustomSeekProviderTest.kt
package org.jellyfin.androidtv.ui.playback.overlay

import io.mockk.every
import io.mockk.mockk
import org.junit.Assert.assertArrayEquals
import org.junit.Assert.assertEquals
import org.junit.Test

class CustomSeekProviderTest {
	@Test
	fun testGetSeekPositions_withSimpleDuration() {
		val videoPlayerAdapter = mockk<VideoPlayerAdapter>()
		every { videoPlayerAdapter.canSeek() } returns true
		every { videoPlayerAdapter.duration } returns 90000L
		val customSeekProvider = CustomSeekProvider(videoPlayerAdapter)
		val expected = longArrayOf(0L, 30000L, 60000L, 90000L)
		val actual = customSeekProvider.seekPositions
		assertArrayEquals(expected, actual)
	}

	@Test
	fun testGetSeekPositions_withOddDuration() {
		val videoPlayerAdapter = mockk<VideoPlayerAdapter>()
		every { videoPlayerAdapter.canSeek() } returns true
		every { videoPlayerAdapter.duration } returns 130000L
		val customSeekProvider = CustomSeekProvider(videoPlayerAdapter)
		val expected = longArrayOf(0L, 30000, 60000, 90000, 120000, 130000)
		val actual = customSeekProvider.seekPositions
		assertArrayEquals(expected, actual)
	}

	@Test
	fun testGetSeekPositions_withSeekDisabled() {
		val videoPlayerAdapter = mockk<VideoPlayerAdapter>()
		every { videoPlayerAdapter.canSeek() } returns false
		val customSeekProvider = CustomSeekProvider(videoPlayerAdapter)
		val actual = customSeekProvider.seekPositions
		assertEquals(0, actual.size.toLong())
	}
}
